infrastructure/src/main/java/org/corfudb/infrastructure/LayoutServer.java

```diff
@@ -257,7 +248,7 @@ public synchronized void handleMessageLayoutPropose(
         // In addition, if the rank in the propose message is equal to the current phase 2 rank
         // (already accepted message), reject.
         // This can happen in case of duplicate messages.
-        if (phase2Rank != null && proposeRank.equals(phase2Rank)) {
+        if (proposeRank.equals(phase2Rank)) {
```

Comment:
Could you explain this change ?

Reply:
checking the phase2Rank not being null is redundant considering that null phase2Rank won't enter the if block because of the remaining part of condition clause.